benches/bench.rs

```diff
@@ -0,0 +1,23 @@
+extern crate core;
+
+use criterion::{criterion_group, criterion_main, Criterion};
+use ff::Field;
+use pasta_curves::Fp;
+
+use spartan2::spartan::polys::eq::EqPolynomial;
+
+fn benchmarks_evaluate_incremental(c: &mut Criterion) {
+  let mut group = c.benchmark_group("evaluate_incremental");
+  (1..=20).step_by(2).for_each(|i| {
+    group.bench_with_input(format!("2^{}", i), &i, |b, &i| {
+      let eq_polynomial =
+        EqPolynomial::new(vec![Fp::random(&mut rand::thread_rng()); 2usize.pow(i)]);
+      b.iter(|| {
+        eq_polynomial.evaluate(vec![Fp::random(&mut rand::thread_rng()); 2usize.pow(i)].as_slice())
```

Comment:
You're creating the vector (which includes memory allocation as well as using an rng) as part of the benchmarked code - I think have a feeling that for the large enough size of the poly, this would incur a non-negligible setup cost which doesn't belong in the benchmarking of the `eval` function.

Reply:
Actually, the point is only `n` in size, while the poly has `2^n` elements, so maybe the opposite is true: this is amortized for large polys, but for small polys, the overhead of allocating an element is large. In either case, the argument still stands: the eval point should not be created as part of the benchmarked code.

Reply:
You are right; I'll fix it in the next commit. The cost is definitely non-negligible for small polys, I think the benchmarks should be better once I do the modifications.